recipes/backward-cpp/all/conanfile.py

```diff
@@ -57,23 +60,33 @@ def config_options(self):
             self.options.stack_details = "backtrace_symbol"
 
     def configure(self):
-        if self.options.shared:
+        if self.options.header_only:
+            self.options.rm_safe("fPIC")
+            self.options.rm_safe("shared")
+        if self.options.get_safe("shared"):
             self.options.rm_safe("fPIC")
 
     def layout(self):
-        cmake_layout(self, src_folder="src")
+        if self.options.header_only:
+            basic_layout(self, src_folder="src")
+        else:
+            cmake_layout(self, src_folder="src")
+
+    def package_id(self):
+        if self.info.options.header_only:
+            self.info.clear()
 
     def requirements(self):
-        if self.settings.os in ["Linux", "Android"]:
+        if self.settings.os in ["Linux", "FreeBSD", "Android"]:
             if self._has_stack_walking("libunwind"):
                 self.requires("libunwind/1.6.2", transitive_headers=True)
             if self._has_stack_details("dwarf"):
-                self.requires("libdwarf/20191104", transitive_headers=True, transitive_libs=True)
+                self.requires("libdwarf/20191104", transitive_headers=True)
```

Comment:
After checking locally, this seems to be needed for shared=True, so I wonder if the `dw` and `bfd` options are also neded. I can't test `dw` because `elfutils` is not Conan 2 ready yet (Waiting on https://github.com/conan-io/conan-center-index/pull/17272), and for `bfd`, I get a missing header compilation error:
```
In file included from /root/.conan2/p/b/backw4b891165fe452/b/src/backward.cpp:36:
/root/.conan2/p/b/backw4b891165fe452/b/src/backward.hpp:229:10: fatal error: bfd.h: No such file or directory
  229 | #include <bfd.h>
      |          ^~~~~~~
```

Reply:
Interesting. As far as I understand it, `transitive_libs=True` is set by default anyway as long as `libs=True` is set (also by default). Is there something I'm missing? I would love to know more, if that's the case.

Reply:
@valgur Sorry I missed this conversation: In Conan 1, it all linked in the end due to how Conan worked back then afaik, but yeah Conan 2 does not transitively make the libs available unless told to otherwise (Except for the case of `package_type=header-library` which has both transitive traits set to true)